pkg/operator/certrotation/client_cert_rotation_controller.go

```diff
@@ -44,23 +70,22 @@ type CertRotationController struct {
 	RotatedSelfSignedCertKeySecret RotatedSelfSignedCertKeySecret
 
 	// Plumbing:
-	OperatorClient v1helpers.StaticPodOperatorClient
+	StatusReporter StatusReporter
 }
 
 func NewCertRotationController(
 	name string,
 	rotatedSigningCASecret RotatedSigningCASecret,
 	caBundleConfigMap CABundleConfigMap,
 	rotatedSelfSignedCertKeySecret RotatedSelfSignedCertKeySecret,
-	operatorClient v1helpers.StaticPodOperatorClient,
 	recorder events.Recorder,
+	reporter StatusReporter,
 ) factory.Controller {
 	c := &CertRotationController{
-		name:                           name,
 		rotatedSigningCASecret:         rotatedSigningCASecret,
 		CABundleConfigMap:              caBundleConfigMap,
 		RotatedSelfSignedCertKeySecret: rotatedSelfSignedCertKeySecret,
-		OperatorClient:                 operatorClient,
+		StatusReporter:                 reporter,
```

Comment:
@stevekuznetsov  why don't we set the conditionName of the reporter to name here? The current approach makes it pretty weird in consumers since we essentially have to duplicate the controllerName when instantiating a controller. For example here: https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/pkg/operator/certrotationcontroller/certrotationcontroller.go#L131-L168, we have to do something like:

```diff
diff --git a/pkg/operator/certrotationcontroller/certrotationcontroller.go b/pkg/operator/certrotationcontroller/certrotationcontroller.go
index 676a26182..f6f18e4b2 100644
--- a/pkg/operator/certrotationcontroller/certrotationcontroller.go
+++ b/pkg/operator/certrotationcontroller/certrotationcontroller.go
@@ -163,8 +163,8 @@ func newCertRotationController(
                        Client:        kubeClient.CoreV1(),
                        EventRecorder: eventRecorder,
                },
-               operatorClient,
                eventRecorder,
+               certrotation.StatusReporter{componentName: "AggregatorProxyClientCert", OperatorClient: operatorClient},
        )
        ret.certRotators = append(ret.certRotators, certRotator)
```

Reply:
So we'd want to refactor `NewCertRotationController` to take a reporter factory or something and close over passing in a name for construction? Sounds reasonable - but we can't assume the reporter impl, is all.

Reply:
@stevekuznetsov @dgrisonnet  I have run into this when bumping the library-go as well. What about this approach? https://github.com/openshift/library-go/pull/1630

Reply:
Your approach seems fine to me. The controller will always know which condition to report and I don't think we have a use-case where we would need one reporter per condition.

@stevekuznetsov would that approach perhaps also fit your use-case?

Reply:
Yep - either let's do that or just export the field, either one is good!